huaweicloud/services/elb/data_source_huaweicloud_elb_security_policies.go

```diff
@@ -0,0 +1,413 @@
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"listeners": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"protocols": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"ciphers": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"updated_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+	return &sc
+}
+
+func dataSourceElbSecurityPoliciesRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	v := d.Get("type").(string)
+	if v == "system" {
+		return systemElbSecurityPoliciesRead(d, meta)
+	} else if v == "custom" {
+		return customElbSecurityPoliciesRead(d, meta)
+	}
+	return allElbSecurityPoliciesRead(d, meta)
+}
+
+func systemElbSecurityPolicies(d *schema.ResourceData, meta interface{}) (interface{}, error) {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	var (
+		listSystemSecurityPoliciesHttpUrl = "v3/{project_id}/elb/system-security-policies"
+		listSystemSecurityPoliciesProduct = "elb"
+	)
+	listSystemSecurityPoliciesClient, err := cfg.NewServiceClient(listSystemSecurityPoliciesProduct, region)
+	if err != nil {
+		return nil, fmt.Errorf("error creating ELB client: %s", err)
+	}
+	listSystemSecurityPoliciesPath := listSystemSecurityPoliciesClient.Endpoint + listSystemSecurityPoliciesHttpUrl
+	listSystemSecurityPoliciesPath = strings.ReplaceAll(listSystemSecurityPoliciesPath, "{project_id}",
+		listSystemSecurityPoliciesClient.ProjectID)
+	getSystemSecurityPoliciesOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	listSystemSecurityPoliciesResp, err := listSystemSecurityPoliciesClient.Request("GET",
+		listSystemSecurityPoliciesPath, &getSystemSecurityPoliciesOpt)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving ELB security policies")
+	}
+	listSystemSecurityPoliciesRespBody, err := utils.FlattenResponse(listSystemSecurityPoliciesResp)
+	if err != nil {
+		return nil, fmt.Errorf("error returns the api response body: %s", err)
+	}
+	return listSystemSecurityPoliciesRespBody, nil
+}
+
+func customElbSecurityPolicies(d *schema.ResourceData, meta interface{}) (interface{}, error) {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	var (
+		listSecurityPoliciesHttpUrl = "v3/{project_id}/elb/security-policies"
+		listSecurityPoliciesProduct = "elb"
```

Comment:
```suggestion
		listCustomSecurityPoliciesHttpUrl = "v3/{project_id}/elb/security-policies"
		listCustomSecurityPoliciesProduct = "elb"
```
and fix other places

Reply:
done